lib/OpenQA/WebAPI/Controller/Admin/User.pm

```diff
@@ -39,17 +43,45 @@ sub update {
         $is_operator = 1;
     }
 
+    my $data = {is_admin => $is_admin, is_operator => $is_operator};
+    $data->{username} = $username if (defined($username));
+    $data->{email} = $email if (defined($email));
+    $data->{fullname} = $fullname if (defined($fullname));
+    $data->{nickname} = $nickname if (defined($nickname));
```

Comment:
I suppose that at least the OpenID auth plugin will override these fields with information from the OpenID provider. So there's no real use in changing them because the change will likely be reverted on the next login anyways.